app/dot/cpp/core/services/LoginService.java

```diff
@@ -156,24 +143,33 @@ private Jws<Claims> getJwsClaims(String jwtToken) throws LoginException {
     }
   }
 
-  public User authorizeRequest(String accessToken, List<UserRole> permittedUserRoles)
+  /**
+   * Authorizes a request using a JWT access token. Verifies token validity and expiration, and
+   * user's role.
+   *
+   * @param accessToken the JWT access token
+   * @param permittedUserRoles a list of authorized user roles; if it's empty, all users are allowed
+   * @return the user ID associated with the validated JWT access token
+   * @throws LoginException If the access token is invalid or expired, or if the user's role does
+   *     not match the permitted roles, a LoginException is thrown.
+   */
+  public String authorizeRequest(String accessToken, List<UserRole> permittedUserRoles)
       throws LoginException {
-    final String userId = checkJwtAndGetUserId(accessToken);
-    try {
-      final var user = userService.findById(userId);
-      logger.debug("{}", user);
+    logger.debug("JWT access token {}", accessToken);
 
-      if (!user.isActive()) {
-        throw LoginException.from(ErrorCodes.USER_INACTIVE_ACCOUNT);
-      }
-      if (!permittedUserRoles.isEmpty() && !permittedUserRoles.contains(user.getRole())) {
-        throw LoginException.from(ErrorCodes.USER_ROLE_MISMATCH);
-      }
+    final var claims = getJwsClaims(accessToken).getBody();
+    final var expirationDate = claims.getExpiration();
 
-      return user;
-    } catch (BaseException e) {
-      throw LoginException.from(ErrorCodes.USER_NOT_FOUND);
+    if (expirationDate.before(new Date())) {
+      throw LoginException.from(ErrorCodes.EXPIRED_ACCESS);
     }
+
+    final var userRole = claims.get(USER_ROLE, String.class);
+    if (!permittedUserRoles.isEmpty() && !permittedUserRoles.contains(UserRole.valueOf(userRole))) {
```

Comment:
shouldn't we also add ADMIN to Authentication of editPage and edit from UserController?

Reply:
I believe setting the permittedUserRoles as default to only USER would be more appropriate, then the check for empty would not be needed. What do you think?

Reply:
> shouldn't we also add ADMIN to Authentication of editPage and edit from UserController?

They should've had that, that's an issue.

Reply:
> I believe setting the permittedUserRoles as default to only USER would be more appropriate, then the check for empty would not be needed. What do you think?

Sure, sounds good.